nodeEditor/src/sharedComponents/checkBoxLineComponent.tsx

```diff
@@ -69,15 +70,22 @@ export class CheckBoxLineComponent extends React.Component<ICheckBoxLineComponen
         this.setState({ isSelected: !this.state.isSelected });
     }
 
+    getDisabled() {
+        if (this.props.disabled) {
+            return this.props.disabled;

```

Comment:
```suggestion
            return !!this.props.disabled;
```
But overall why this function?

Reply:
I added an optional property `disabled` to `CheckBoxLineComponent` so that I know when the checkbox should not be responsive to user input.

Specifically, for node ports that are connected to outside nodes, I can still show the checkbox and also communicate that connected nodes are automatically exposed on the frame.

Connected node port (auto-disabled checkbox)
![Screen Shot 2020-04-22 at 12 19 56 AM](https://user-images.githubusercontent.com/25019989/79952338-1d697000-842f-11ea-8cd2-a2b362031ae9.png)

Not connected node port (checkbox enabled)
![Screen Shot 2020-04-22 at 12 20 21 AM](https://user-images.githubusercontent.com/25019989/79952377-3114d680-842f-11ea-9ea6-6da3be05bf89.png)

I wrote the function that way because I wanted to make sure that past imports of `CheckBoxLineComponent` wouldn't be affected by the addition of the `disabled` property. But I see now if I do the `!!`, it should accomplish the same thing. Updated.